### PR TITLE
Save-out-regional-results-in-calendar-years

### PR DESCRIPTION
**Pull request by liviajakob**
Super small PR, just saving out CSVs of each region in calendar years besides only hydrological years (as requested by the GlaMBIE consortium).
Edit: Also sorted out some of the deprecation warnings, so Alex won't feel like he's reviewing 2 lines of code only;)

### CODE REVIEW

File: glambie/data/data_catalogue.py [hunk 1]
```diff
@@ -310,22 +309,25 @@ def average_timeseries_in_catalogue(self, remove_trend: bool = True, add_trend_a
 
         # join all catalogues by start and end dates
         # the resulting dataframe has a set of columns with repeating prefixes
-        df = reduce(lambda left, right: left.merge(right, how="outer", on=["start_dates", "end_dates"]), catalogue_dfs)
-        df = df.sort_values(by="start_dates")
-        start_dates, end_dates = np.array(df["start_dates"]), np.array(df["end_dates"])
-        mean_changes = np.array(df[df.columns.intersection(
-            df.filter(regex=("changes*")).columns.to_list())].mean(axis=1))
+        df_merged_all = pd.concat([x.set_index(['start_dates', 'end_dates']) for x in catalogue_dfs],
```
Comment (alex-ew) [inline glambie/data/data_catalogue.py:312]: Minor one - leaving this simply as "df" is fine. Or at the very least, the last _all is a tautology, so df_merged or df_all would read better.

File: glambie/processing/process_global_results.py [hunk 2]
```diff
@@ -86,6 +85,14 @@ def run_global_results(glambie_run_config: GlambieRunConfig,
             region=REGIONS["global"], data_group=GLAMBIE_DATA_GROUPS["consensus"], csv_file_name="global_gt.csv")
         global_timeseries_gt.save_data_as_csv(csv_output_file)
 
+        # save out regional results homogenized to calendar year (in mwe and in gt)
```
Comment (alex-ew) [inline glambie/processing/process_global_results.py:88]: Very minor. Comment on line 57 should reflect plot and save to csv

File: glambie/processing/process_global_results.py [hunk 3]
```diff
@@ -86,6 +85,14 @@ def run_global_results(glambie_run_config: GlambieRunConfig,
...
+        for results in zip(regional_results_catalogue_homogenized.datasets,
+                           regional_results_catalogue_homogenized_gt.datasets):
+            for result in results:
+                result.save_data_as_csv(output_path_handler.get_csv_output_file_path(
+                    region=result.region, data_group=GLAMBIE_DATA_GROUPS["consensus"],
+                    csv_file_name=f"consensus_calendar_year_{result.unit.lower()}_{result.region.name}.csv"))
```
Comment (alex-ew) [inline glambie/processing/process_global_results.py:94]: Would be nice to see a test to make sure this file is created / valid csv

**Review by alex-ew — COMMENTED** [review]
Looks good. I think the most important change would be testing the csv creation. The other comments are minor observations.

File: glambie/data/data_catalogue.py (hunk 1 above)
Comment (liviajakob) [reply glambie/data/data_catalogue.py:312]: done

File: glambie/processing/process_global_results.py (hunk 2 above)
Comment (liviajakob) [reply glambie/processing/process_global_results.py:88]: done

File: glambie/processing/process_global_results.py (hunk 3 above)
Comment (liviajakob) [reply glambie/processing/process_global_results.py:94]: done

**Comment by liviajakob** [comment]
Thanks for review! I addressed all comments.

File: glambie/processing/process_global_results.py (hunk 3 above)
Comment (alex-ew) [reply glambie/processing/process_global_results.py:94]: Nice. That'll do

**Review by alex-ew — APPROVED** [review]
Nice - thanks

**Pull request merged**